packages/assets-controllers/src/assetsUtil.test.ts

```diff
@@ -436,4 +438,83 @@ describe('assetsUtil', () => {
       expect(timestampsIncreasing).toBe(true);
     });
   });
+
+  describe('fetchAndMapExchangeRates', () => {
+    const mockedPricesService: AbstractTokenPricesService = {
+      validateChainIdSupported(_chainId: unknown): _chainId is Hex {
+        return true;
+      },
+      validateCurrencySupported(_currency: unknown): _currency is string {
+        return true;
+      },
+      async fetchTokenPrices() {
+        return {};
+      },
+    };
+
+    it('should return empty object when chainId not supported', async () => {
+      const testTokenAddress = '0x7BEF710a5759d197EC0Bf621c3Df802C2D60D848';
+
+      jest
+        .spyOn(mockedPricesService, 'validateChainIdSupported')
+        .mockReturnValue(false);
+
+      const result = await assetsUtil.fetchTokenContractExchangeRates({
+        tokenPricesService: mockedPricesService,
+        nativeCurrency: 'ETH',
+        tokenAddresses: [testTokenAddress],
+        chainId: '0x1',
+      });
+
+      expect(result).toStrictEqual({});
+    });
+
+    it('should return empty object when nativeCurrency not supported', async () => {
+      const testTokenAddress = '0x7BEF710a5759d197EC0Bf621c3Df802C2D60D848';
+      jest
+        .spyOn(mockedPricesService, 'validateCurrencySupported')
+        .mockReturnValue(false);
+
+      const result = await assetsUtil.fetchTokenContractExchangeRates({
+        tokenPricesService: mockedPricesService,
+        nativeCurrency: 'X',
+        tokenAddresses: [testTokenAddress],
+        chainId: '0x1',
+      });
+
+      expect(result).toStrictEqual({});
+    });
+
+    it('should return successfully', async () => {
+      const testTokenAddress = '0x7BEF710a5759d197EC0Bf621c3Df802C2D60D848';
+      const testNativeCurrency = 'ETH';
+      const testChainId = '0x1';
+      jest
+        .spyOn(mockedPricesService, 'validateCurrencySupported')
+        .mockReturnValue(true);
+
+      jest
+        .spyOn(mockedPricesService, 'validateChainIdSupported')
+        .mockReturnValue(true);
+
+      jest.spyOn(mockedPricesService, 'fetchTokenPrices').mockResolvedValue({
+        [testTokenAddress]: {
+          tokenAddress: testTokenAddress,
+          value: 0.0004588648479937523,
+          currency: testNativeCurrency,
+        },
+      });
+
+      const result = await assetsUtil.fetchTokenContractExchangeRates({
+        tokenPricesService: mockedPricesService,
+        nativeCurrency: testNativeCurrency,
+        tokenAddresses: [testTokenAddress],
+        chainId: testChainId,
+      });
+
+      expect(result).toMatchObject({
+        [testTokenAddress]: expect.any(Number),
```

Comment:
Since we control the mock data, should we be explicit here?

``` suggestion
        [testTokenAddress]: 0.0004588648479937523,
```

Reply:
Definitely!